### DISCUSSION TITLE
Трекинг головы

### DISCUSSION THREAD

**Discussion by lazenyuk-dmitry** [vsulako/AFFBWheel#22] [General]
Всем привет. Задался я тут вопросом трекинга положения головы в играх. Был у меня уже опыт с бич-VR) Я использовал китайские очки для телефона, trinus vr собственно для vr в играх и open track для отслеживания положения головы и ReShade для создания глубины 3D в играх в которых нет VR. Это конечно прикольно но жрет половину FPS, датчики не особо точные и поэтому плывет иногда трекинг положения головы и у этого всего есть задержки, поэтому играть не очень комфортно так, но прикольно.

Потом я наткнулся на **[видосик на ютубе](https://www.youtube.com/watch?v=mgaKApshWEo)** там парень рассказывает про все виды трекинга и делает вывод что лучше всего FreeTrack со светодиодными лампами и веб камерой. Но я так посмотрел подумал. Саму приблуду со светодиодами сделать не проблема и не дорого. Но вот вебка хорошая стоит нормально так и еще нужно ее разобрать и светофильтр установить для лучшего функционирования. Так подумал и что то забросил эту идею.

А сегодня подумал что может быть есть варианты на ардуино и не дорого и загуглил и вот решение **[Инерционный трекер на базе Arduino и GY-85](https://www.youtube.com/watch?v=XeHk2mNYr5A)** на мой взгляд стоит чтобы повторить. Вот только нужно бы посмотреть еще как это дело сделать беспроводным и было бы идеально. @vsulako, можешь подсказать как лучше это сделать и ардуино и чип все нужно как в родике? У меня просто есть в наличии ProMicro. Ну и может у кого есть другие идеи или датчик героскопа интереснее можно найти? Или у кого был какой опыт в этом деле и кто то знает другие простые и не дорогие решения? Хотел бы обсудить эту тему.

**Comment by lazenyuk-dmitry** [comment]
Нашел пару интересных ссылок. Разбираюсь.

https://github.com/TrueOpenVR/TrueOpenVR-Drivers/blob/master/README.RU.md

https://github.com/Razor-AHRS/razor-9dof-ahrs

**Reply by vsulako** [reply]
Я вот на чего натыкался: https://sites.google.com/site/diyheadtracking/
Как раз там беспроводной вариант есть, по WiFi или Bluetooth.
Все нюансы расписаны, датчиков куча поддерживается.

**Comment by lazenyuk-dmitry** [comment]
Можно вместо GY-85 использовать MPU 6050 что еще намного дешевле получается.

**Comment by mitiaborodada** [comment]
Я как раз полностью реализовал это решение -  **[Инерционный трекер на базе Arduino и GY-85](https://www.youtube.com/watch?v=XeHk2mNYr5A)**
Пока доволен. Со временем может быть передалаю в беспроводное. Единственный минус - это то, что нужно использовать сторонюю программу для такого трекера.

**Comment by lazenyuk-dmitry** [comment]
> Я как раз полностью реализовал это решение -  **[Инерционный трекер на базе Arduino и GY-85](https://www.youtube.com/watch?v=XeHk2mNYr5A)**
> Пока доволен. Со временем может быть передалаю в беспроводное. Единственный минус - это то, что нужно использовать сторонюю программу для такого трекера. 
> 

Если я не ошибаюсь GY-85 не поддерживает приближение? Не все оси.

**Reply by lazenyuk-dmitry** [reply]
А там просто второй датчик нужен на шею или плечи крепить. Тогда будет еще две оси отслеживаться. И тогда будет трекаться приближение/удаление от монитора и смещение влево и право. Трекинка вверх и вниз я так понял не реализован.

**Reply by mitiaborodada** [reply]
А какой смысл в приближении и удалении? Мне кажется, в симуляции особого значения не имеют.

**Reply by lazenyuk-dmitry** [reply]
Ну для авто симуляторов это да, не нужно. А вот для авио симуляторов там нужно приближение. Это так к слову) я тоже буду делать 3dof а не 5dof трекер.
